design-proposals/klip-61-bootstrapped-table-subscriptions.md

```diff
@@ -0,0 +1,85 @@
+# KLIP 61 - Bootstrapped TABLE subscriptions
+
+**Author**: Natea Eshetu Beshada @nateab |
+**Release Target**: 0.27.0 |
+**Status**: _In Discussion_ |
+**Discussion**: _link to the design discussion PR_
+
+**tl;dr:** We want users to be able to very easily with a single query get the current state from a `TABLE` and 
+simultaneously subscribe to any future changes to that `TABLE` by specifying `EMIT ALL`.
+
+## Motivation and background
+
+ksqlDB is a streaming database that allows users to work with data in motion (stream processing) and data at rest (databases).
+Thus, we have push queries which let you subscribe to a `TABLE` as it changes in real-time, and also pull 
+queries which retrieves the current state of the table, similar to a traditional database query. To further cement ksqlDB's
+positioning at the intersection of stream processing and traditional databases, it is necessary to develop features that 
+seamlessly combine the two. Adding bootstrapped `TABLE` subscriptions to ksqlDB achieves this purpose by allowing users to query their 
+data at rest up until now, while also subscribing to any later changes from data that is in motion. 
+
+Users can try to achieve this functionality by simply combining a `TABLE` pull query followed by a `TABLE` push query, but this 
+could result in gaps in the returned data since there are no guarantees that the push query will not miss any data after the 
+pull query is executed. Implementing this feature, we will keep track off the offsets under the hood for the user so that 
+we can guarantee there is no missing data. It will also be more clean and intuitive for users to have to only use one query
+to achieve this.
+
+## What is in scope
+
+- Queries should be supported on `TABLES` with `EMIT ALL` at the end.
+- Bootstrapped subscriptions should retrieve all past data and subscribe to any further changes.
```

Comment:
> all past data

Sound like historic updates? Should we say all "current rows in the TABLE" instead?

Reply:
Yeah I agree this can be clearer, will update to say "current rows in the TABLE"